src/tests/.clang-format

```diff
@@ -0,0 +1,49 @@
+---
```

Comment:
Can we move this file to the root of the repository.

Reply:
So I don't know that much about clang-format do you know if it will pick up on .clang-format if it is not in the cwd?

Reply:
It will go through all parent directories until it find one.